src/hooks/loadables/useLoadSpendingLimits.ts

```diff
@@ -73,12 +120,27 @@ export const useLoadSpendingLimits = (): AsyncResult<SpendingLimitState[]> => {
   const { safeAddress, safe, safeLoaded } = useSafeInfo()
   const chainId = useChainId()
   const provider = useWeb3ReadOnly()
+  const { balances } = useBalances()
+
+  const tokenInfoFromBalances = useMemo(() => {
+    if (balances.items.length === 0) {
+      return undefined
+    }
+    return balances.items.reduce((dictionary, item) => {
+      dictionary[item.tokenInfo.address] = {
+        symbol: item.tokenInfo.symbol,
+        decimals: item.tokenInfo.decimals,
+        logoUri: item.tokenInfo.logoUri,
+      }
+      return dictionary
+    }, {} as TokenInfoFromBalances)
+  }, [balances.items])
```

Comment:
```suggestion
  const tokenInfoFromBalances = useMemo(() => {
    if (balances.items.length === 0) {
      return undefined
    }
    return balances.items.reduce<TokenInfoFromBalances>((dictionary, item) => {
      dictionary[item.tokenInfo.address] = {
        symbol: item.tokenInfo.symbol,
        decimals: item.tokenInfo.decimals,
        logoUri: item.tokenInfo.logoUri,
      }
      return dictionary
    }, {})
  }, [balances.items])
```

Reply:
We could alternatively create a selector that does this for us. What do you think?